Civi/Eck/API/Entity.php

```diff
@@ -163,6 +166,41 @@ public static function getEckAfforms($event) {
         }
         $afforms[$name] = $item;
       }
+
+      // Search listing for for each type
+      $name = 'afsearch' . $entityType['entity_name'] . '_listing';
+      $item = [
+        'name' => $name,
+        'type' => 'search',
+        'title' => $entityType['label'],
+        'description' => E::ts('Search listing for %1', [1 => $entityType['label']]),
+        'is_dashlet' => FALSE,
+        'is_public' => FALSE,
+        'is_token' => FALSE,
+        'permission' => 'access CiviCRM',
+        'server_route' => "civicrm/eck/entity/list/{$entityType['name']}",
+        'requires' => ['crmSearchDisplayTable'],
+      ];
+      $item['layout'] = "<div af-fieldset=\"\">\n";
+      $item['layout'] .= "  <div class=\"pull-right btn-group\">\n";
+      $item['layout'] .= "    <button type=\"button\" class=\"btn dropdown-toggle btn-primary\" crm-icon=\"fa-plus\" data-toggle=\"dropdown\" aria-haspopup=\"true\" aria-expanded=\"false\">\n";
+      $item['layout'] .= "      " . E::ts('Add %1', [1 => $entityType['label']]) . " <span class=\"caret\"></span>\n";
+      $item['layout'] .= "    </button>\n";
+      $item['layout'] .= "    <ul class=\"dropdown-menu\">\n";
+      foreach ($subTypes as $subType) {
+        $item['layout'] .= "      <li>\n";
+        $item['layout'] .= "        <a href=\"{{:: crmUrl('civicrm/eck/entity/edit/{$entityType['name']}/{$subType['name']}') }}\">{$subType['label']}</a>\n";
+        $item['layout'] .= "      </li>\n";
+      }
+      $item['layout'] .= "    </ul>\n";
+      $item['layout'] .= "  </div>\n";
+      $item['layout'] .= "  <div class=\"af-container af-layout-inline\">\n";
+      $item['layout'] .= "    <af-field name=\"title\" defn=\"{required: false, input_attrs: {placeholder: '" . E::ts('Filter by Title') . "'}, label: false}\" />\n";
+      $item['layout'] .= "    <af-field name=\"subtype\" defn=\"{input_type: 'Select', input_attrs: {multiple: true, placeholder: '" . E::ts('Filter by Type') . "'}, required: false, label: false}\" />\n";
+      $item['layout'] .= "  </div>\n";
+      $item['layout'] .= "  <crm-search-display-table search-name=\"ECK_Listing_" . $entityType['name'] . "\" display-name=\"ECK_Listing_Display" . $entityType['name'] . "\"></crm-search-display-table>\n";
+      $item['layout'] .= "</div>\n";
```

Comment:
Can the Search display's template live in a separate file, in order to not have this pile of HTML strings in here?